yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -390,8 +393,22 @@ private[spark] class Client(
     // Upload Spark and the application JAR to the remote file system if necessary,
     // and add them as local resources to the application master.
     val fs = destDir.getFileSystem(hadoopConf)
-    val nns = YarnSparkHadoopUtil.get.getNameNodesToAccess(sparkConf) + destDir
-    YarnSparkHadoopUtil.get.obtainTokensForNamenodes(nns, hadoopConf, credentials)
+
+    // Merge credentials obtained from registered providers
+    var nearestTimeOfNextRenewal = Long.MaxValue
+    credentialManager.obtainCredentials(hadoopConf, credentials).foreach { t =>
```

Comment:
You don't really need to calculate `nearestTimeOfNextRenewal` if `loginFromKeytab` is false, right?

Also, this all could be:

```
val nearestTimeOfNextRenewal =  credentialManager.obtainCredentials(hadoopConf, credentials).flatten.min
```

Just be careful to:
- validate that all returned times are after the current time
- watch out for when the result of `.flatten` is empty